src/sampleutil.cpp

```diff
@@ -178,8 +178,14 @@ void SampleUtil::copyWithRampingGain(CSAMPLE* pDest, const CSAMPLE* pSrc,
 // static
 void SampleUtil::convertS16ToFloat32(CSAMPLE* pDest, const SAMPLE* pSrc,
         unsigned int iNumSamples) {
+    // Sample min can be -32768 instead of -32767, so make sure our conversion
+    // scale is picking the largest possible absolute value.
+    // Note that this means that although some sample values convert to -1.0,
+    // none will convert to +1.0.
+    static const CSAMPLE kConversionFactor = math_max(CSAMPLE(-SAMPLE_MIN),
```

Comment:
In the first version 32768 is a constant while in the second version it checks if it's already done the static initialization on every call.